File: .leetcode/26.删除排序数组中的重复项.go
/*
 * @lc app=leetcode.cn id=26 lang=golang
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start
func removeDuplicates(nums []int) int {
	numLen := len( nums )
    if numLen <= 1{
		return numLen
	}
	cursor := 0
	for index := 1; index < numLen; index++ {
		if nums[cursor] != nums[index] {
			cursor++
			if cursor != index {
				nums[cursor] = nums[index]
			}
		}
	}  
	return cursor+1
}
// @lc code=end

File: .leetcode/26.删除排序数组中的重复项.js
/*
 * @lc app=leetcode.cn id=26 lang=javascript
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if( nums.length === 0 ) return 0;

    let cursor = 0;
    nums.forEach( ( num )=>{
        if( nums[cursor] != num){
            nums[++cursor] = num;
        }
    } )

    return cursor+1;
};
// @lc code=end

